src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs

```diff
@@ -372,28 +404,37 @@ private unsafe void ParseEventBufferAndNotifyForEach(ReadOnlySpan<byte> buffer)
         /// State information used by the ReadDirectoryChangesW callback.  A single instance of this is used
         /// for an entire session, getting passed to each iterative call to ReadDirectoryChangesW.
         /// </summary>
-        private sealed class AsyncReadState
+        private sealed class AsyncReadState : IDisposable
```

Comment:
Does this actually need to implement `IDisposable`? I do not see it used anywhere, and implements the Dispose contract slightly incorrectly (Dispose does not support being called multiple times).

Reply:
No, it could be removed.